homeassistant/components/fronius/quality_scale.yaml

```diff
@@ -0,0 +1,87 @@
+rules:
+  # Bronze
+  action-setup:
+    status: exempt
+    comment: |
+      This integration does not provide additional actions.
+  appropriate-polling: done
+  brands: done
+  common-modules:
+    status: done
+    comment: |
+      Single platform only, so no entity.py file.
+      CoordinatorEntity is used.
+  config-flow-test-coverage: done
```

Comment:
All tests should end in either a CREATE_ENTRY or ABORT so we can test that the flow is able to end.

Reply:
There are a bunch of tests here that end in 
```py
    assert result2["type"] is FlowResultType.FORM
    assert result2["errors"] == {"base": "cannot_connect"}
```
The idea of this is to test error conditions. Ie. when a user mistypes the IP address I'd prefer not to abort, but to show an error and let the user correct the input.

The flow consists of a single step - so this is well tested to be able to end in other tests scoped for a successful outcome.

Reply:
But we don't test if the flow is able to recover from these kind of errors. We want to avoid introducing a race bug, so ideally we make the tests end to end